### PR TITLE
feat: update Windows base images to 2024.1B

### PR DESCRIPTION
**Pull request by AbelHu**
**What type of PR is this?**
/kind feature

**Which issue(s) this PR fixes**:
Update Windows base images to 2024.1B
- Windows 2019: [January 9, 2024—KB5034127 (OS Build 17763.5329)](https://support.microsoft.com/en-us/topic/january-9-2024-kb5034127-os-build-17763-5329-4de58ce5-eb0d-4b9a-95d1-aa15fe30b082)
- Windows 2022: [January 9, 2024—KB5034129 (OS Build 20348.2227)](https://support.microsoft.com/en-us/topic/january-9-2024-kb5034129-os-build-20348-2227-6958a36f-efaf-4ef5-a576-c5931072a89a)

**What this PR does / why we need it**:

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

**Review by ShiqianTao — APPROVED** [review]

**Review by junjiezhang1997 — APPROVED** [review]

File: vhdbuilder/packer/configure-windows-vhd.ps1 [hunk 1]
```diff
@@ -473,6 +473,13 @@ function Update-Registry {
             Write-Log "The current value of 3767762061 is $currentValue"
         }
         Set-ItemProperty -Path "HKLM:\SYSTEM\CurrentControlSet\Policies\Microsoft\FeatureManagement\Overrides" -Name 3767762061 -Value 1 -Type DWORD
+
+        Write-Log "Enable 1 fix in 2024-01B"
+        $currentValue=(Get-ItemProperty -Path "HKLM:\SYSTEM\CurrentControlSet\Policies\Microsoft\FeatureManagement\Overrides" -Name 1102009996 -ErrorAction Ignore)
+        if (![string]::IsNullOrEmpty($currentValue)) {
+            Write-Log "The current value of 1102009996 is $currentValue"
+        }
+        Set-ItemProperty -Path "HKLM:\SYSTEM\CurrentControlSet\Policies\Microsoft\FeatureManagement\Overrides" -Name 1102009996 -Value 1 -Type DWORD
```
Comment (ShiqianTao) [inline vhdbuilder/packer/configure-windows-vhd.ps1:482]: Add test in vhdbuilder/packer/test/windows-vhd-content-test.ps1

File: vhdbuilder/packer/generate-windows-vhd-configuration.ps1 [hunk 2]
```diff
@@ -28,17 +28,17 @@ $global:defenderUpdateInfoUrl = "https://go.microsoft.com/fwlink/?linkid=870379&
 
 switch -Regex ($windowsSku) {
     "2019-containerd" {
-        $global:patchUrls = @("https://catalog.s.download.windowsupdate.com/d/msdownload/update/software/secu/2023/12/windows10.0-kb5033371-x64_5ee5edaf9559ed1da54b3c9522e9b2a8c8c71e59.msu")
-        $global:patchIDs = @("KB5033371")
+        $global:patchUrls = @("https://catalog.s.download.windowsupdate.com/c/msdownload/update/software/secu/2024/01/windows10.0-kb5034127-x64_7ac59afbc502ef9dc77f7584b31110ea0b23a566.msu")
+        $global:patchIDs = @("KB5034127")
```
Comment (ShiqianTao) [inline vhdbuilder/packer/generate-windows-vhd-configuration.ps1:32]: Question: Do we need to update `vhdbuilder/packer/windows-image.env`?
- WINDOWS_2019_BASE_IMAGE_VERSION
- WINDOWS_2022_BASE_IMAGE_VERSION
- ...

File: vhdbuilder/packer/configure-windows-vhd.ps1 (hunk 1 above)
Comment (AbelHu) [reply vhdbuilder/packer/configure-windows-vhd.ps1:482]: Good catch. Missed it in rebasing.

File: vhdbuilder/packer/generate-windows-vhd-configuration.ps1 (hunk 2 above)
Comment (AbelHu) [reply vhdbuilder/packer/generate-windows-vhd-configuration.ps1:32]: Will update them although we have updated the patches.

**Pull request merged**